UPGRADING.md

```diff
@@ -0,0 +1,335 @@
+
+**Valid Resource Names**:
+
+* `"projects/[PROJECT_ID]"`
+* `"organizations/[ORGANIZATION_ID]"`
+* `"billingAccounts/[BILLING_ACCOUNT_ID]"`
+* `"folders/[FOLDER_ID]"`
+
+
+#### `google.cloud.logging_v2.client.Client`
+
+
+> **WARNING**: Breaking change
+
+`list_entries` accepts an optional `resource_names` parameter. `projects` has been removed.
+
+
+```diff
+from google.cloud import logging_v2
+
+client = logging_v2.Client()
+-client.list_entries(projects="myProject"])
++client.list_entries(resource_names=["projects/myProject", "folders/myFolder"])
+client.list_entries()  # defaults to project bound to client
+```
+
+`list_sinks` accepts an optional `parent` parameter.
+
+```py
+from google.cloud import logging_v2
+
+client = logging_v2.Client()
+client.list_sinks()  # lists sinks in current project
+client.list_sinks(parent="folders/myFolder")  # specify a different parent resource
+```
+
+#### `google.cloud.logging_v2.logger.Logger`
+
+> **WARNING**: Breaking change
+
+`list_entries` accepts an optional `resource_names` parameter. `projects` has been removed.
+
+```diff
+from google.cloud import logging_v2
+
+client = logging_v2.Client()
+logger = logging_v2.Logger("myLog", client)
+- logger.list_entries(projects="myProject"])
++ logger.list_entries(resource_names=["projects/myProject", "folders/myFolder"])
+logger.list_entries()  # defaults to project bound to client
+```
+
+#### `google.cloud.logging_v2.sinks.Sink`
+
+> **WARNING**: Breaking change
+* Sinks no longer have a `project` property. The attribute is replaced by `parent`.
+
+```diff
+from google.cloud import logging
+
+client = logging_v2.Client(project="myProject")
+sink = logging.Sink("mySink", client=client)
+-project = sink.project  # myProject
++parent = sink.parent  # projects/myProject
+```
+
+
+### `google.cloud.logging` is an alias for `google.cloud.logging_v2`
+
+> **WARNING**: Breaking change
+
+All library code has been moved to `google.cloud.logging_v2`.
+`google.cloud.logging` serves as a default alias for `google.cloud.logging_v2`.
+
+
+
+
+## Changes in GAPIC layer
+
+This section describes changes in the GAPIC layer (produced by the generator) that previously lived in  `google.cloud.logging_v2.proto` / `google.cloud.logging_v2.gapic`.
+
+> **NOTE**: Most users are unlikely to have been using this layer directly.
+
+### Import path
+
+> **WARNING**: Breaking change
+
+The generated client is no longer exposed at `logging_v2`.
+
+**Before**
+```py
+from google.cloud import logging_v2
+logging_client = logging_v2.LoggingServiceV2Client()
+```
+
+**After**
+
+```py
+from google.cloud.logging_v2.services.logging_service_v2 import LoggingServiceV2Client
+from google.cloud.logging_v2.types import LogSink
+
+logging_client = LoggingServiceV2Client()
+sink = LogSink()
+```
+### Method Calls
+
+> **WARNING**: Breaking change
+
+Methods expect request objects. We provide a script that will convert most common use cases.
```

Comment:
This should only affect users using the gapic layer, right?

Reply:
Yep, I'll add another note to make that more clear.